lib/horizon/transactions.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Stellar.Horizon.Transactions do
+  @moduledoc """
+  Exposes functions to interact with Transactions in Horizon.
+
+  You can:
+  - Create a transaction.
+
+  Horizon API reference: https://developers.stellar.org/api/resources/transactions/
+  """
+
+  alias Stellar.Horizon.Error
+  alias Stellar.Horizon.Resource.Transaction
+  alias Stellar.Horizon.Client, as: Horizon
+
+  @type response :: {:ok, Transaction.t()} | {:error, Error.t()}
+
+  @endpoint "/transactions"
+
+  @spec create(base64_envelope :: String.t()) :: response()
+  def create(base64_envelope) do
+    headers = [{"Content-Type", "application/x-www-form-urlencoded"}]
+    body = URI.encode_query(tx: base64_envelope)
+
+    with {:ok, tx} <- Horizon.request(:post, @endpoint, headers, body) do
+      {:ok, Transaction.new(tx)}
+    end
```

Comment:
I don't get this `with` statement here. Why it is better than a `case`?. Also, what about the errors?

Reply:
Errors are handled directly by the client implementation `Stellar.Horizon.Client.Default`, there are two reasons for this:
- Avoid handling errors on every module that interacts with Horizon. 
- Keep a standardized way to manage errors and remove code duplication.

The `with` statement is just a personal preference since errors are being managed by the client, no error clauses are needed using a `with`, if the condition is not met the not matching response from `Horizon.request/4` is returned.

Long story short .... it is equivalent to:

```elixir
    case Horizon.request(:post, @endpoint, headers, body) do
      {:ok, tx} -> Transaction.new(tx)
      error -> error
    end
```

Reply:
Now ..... if you didn't get it probably the code is not declarative enough 😅 , let me change it!